### PR TITLE
Release (Iteration 2.1)

### PR DESCRIPTION
**Pull request by Matt1994**

### CODE REVIEW

**Review by babeanu-dorian — APPROVED** [review]
Good progress on both front-end and back-end. A couple new bugs which will be addressed in the next iteration.

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      - g++-4.9
+
+services:
+  - mongodb
+
+node_js:
+  - "stable"
+
+notifications:
+  slack:
```
Comment (mavagio) [inline .travis.yml:20]: do you use Slack ? if not this is irrelevant.

File: public/css/style.css [hunk 2]
```diff
@@ -163,17 +162,35 @@ h2.page-header {
 }
 
 /* MEDIA QUERIES */
+@media (min-width: 576px) {
```
Comment (mavagio) [inline public/css/style.css:165]: nice, but doesn't bootstrap naturally deal with this ?

File: public/js/medid/login.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+	
+	// Check if the string contains at least one symbol
+	var containsSymbol 		= !(/^[0-9a-zA-Z]+$/.test(string));
+	
+	// Calculate the 'score' of the password
+    var count = containsNumber + containsUppercase + containsSymbol;
+	
+	// Depending on the count variable, assign different 
+	if(string.length < 8 || count <= 1){
+		passwordstrength.updateStyling('Weak', 'red', 'white');
```
Comment (mavagio) [inline public/js/medid/login.js:42]: Why don't you have css class for the 3 possibilities: weak-password, fair-password, strong-password. You will separate styling from code.

File: routes/users.js [hunk 4]
```diff
@@ -113,27 +96,58 @@ passport.deserializeUser(function(id, done) {
...
+	res.redirect('/login');
 });
 
-function encrypt(text, key){
-  var cipher = crypto.createCipher(algorithm, key)
-  var encrypted = cipher.update(text,'utf8','hex') + cipher.final('hex');
-  return encrypted;
+function genCode() {
+	var LENGTH = 12;
+	var ALPHABET = "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ";
```
Comment (mavagio) [inline routes/users.js:112]: *alphanumeric

File: test/test.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+// require cleanup.js in the test files and call tryCleanup on it in every test before asserting
+cleanup.init(11, function(){
+	console.log('cleanup');
+	server.close();
+	mongoose.disconnect();
+});
+
+var suite = vows.describe('Testing');
+
+// testing modules add batches to the suite object (order is important):
```
Comment (mavagio) [inline test/test.js:48]: /**
 *
*/

File: README.md [hunk 6]
```diff
@@ -4,17 +4,19 @@ Running project: [VIEW](https://medid.herokuapp.com/)
```
Comment (mavagio) [inline README.md:4]: You can now use coveralls https://coveralls.io/ to see code coverage percentage.

File: package.json [hunk 7]
```diff
@@ -23,11 +25,16 @@
...
     "passport": "*",
     "passport-http": "*",
     "passport-local": "*",
     "qrcode-js": "0.0.2",
-    "sharp": "^0.17.3",
-    "nodemailer": "*"
+    "sharp": "^0.17.3"
+  },
+  "devDependencies": {
+    "vows": "*",
```
Comment (mavagio) [inline package.json:37]: It is not always a good idea to have the latest version for dependencies. You usually would want to verify that there are no breaking changes in the dependencies before upgrading.

**Review by mavagio — APPROVED** [review]

**Pull request merged**